webservices/rest/src/main/java/com/epam/rest/dao/UserDaoImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.epam.rest.dao;
+
+
+import com.epam.rest.domain.User;
+import org.springframework.stereotype.Repository;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+
+    private final Map<Integer, User> storage = new HashMap<>();
+    private int counter = 0;
```

Comment:
It is not very efficient way actually. `ConcurrentHashMap` uses sophisticated techniques to reduce the need for synchronization and allow parallel non-blocking read access by multiple threads. `SynchronizedMap` will block threads and degrade performance.